kornia/contrib/extract_patches.py

```diff
@@ -236,51 +235,39 @@ def combine_tensor_patches(
         This function is supposed to be used in conjunction with :func:`extract_tensor_patches`.
     """
 
-    if len(patches.shape) != 5:
+    if patches.ndim != 5:
         raise ValueError(f"Invalid input shape, we expect BxNxCxHxW. Got: {patches.shape}")
 
     original_size = cast(Tuple[int, int], _pair(original_size))
     window_size = cast(Tuple[int, int], _pair(window_size))
     stride = cast(Tuple[int, int], _pair(stride))
-
-    if stride[0] != window_size[0] or stride[1] != window_size[1]:
-        raise NotImplementedError(
-            f"Only stride == window_size is supported. Got {stride} and {window_size}."
-            "Please feel free to drop a PR to Kornia Github."
-        )
-
-    if unpadding:
-        unpadding = cast(PadType, _pair(unpadding))
-
-        if len(unpadding) not in [2, 4]:
-            raise AssertionError("Unpadding must be either an int, tuple of two ints or tuple of four ints")
-
-        if len(unpadding) == 2:
-            pad_vert = _pair(unpadding[0])
-            pad_horz = _pair(unpadding[1])
-        else:
-            pad_vert = unpadding[:2]
-            pad_horz = unpadding[2:]
-        unpadding = cast(Tuple[int, int, int, int], pad_horz + pad_vert)
-
-        hpad_check = (original_size[0] + unpadding[2] + unpadding[3]) % window_size[0] == 0
-        wpad_check = (original_size[1] + unpadding[0] + unpadding[1]) % window_size[1] == 0
-
-        if not hpad_check or not wpad_check:
-            raise NotImplementedError("Insufficient padding")
-
-        window_size = (
-            (original_size[0] + (unpadding[2] + unpadding[3])) // window_size[0],
-            (original_size[1] + (unpadding[0] + unpadding[1])) // window_size[1],
-        )
-
-    patches_tensor = patches.view(-1, window_size[0], window_size[1], *patches.shape[-3:])
-    restored_tensor = concatenate(torch.chunk(patches_tensor, window_size[0], 1), -2).squeeze(1)
-    restored_tensor = concatenate(torch.chunk(restored_tensor, window_size[1], 1), -1).squeeze(1)
-
-    if unpadding:
-        unpadding = cast(Tuple[int, int, int, int], unpadding)
-        restored_tensor = pad(restored_tensor, [-i for i in unpadding])
+    unpadding = cast(Tuple[int, int], _pair(unpadding))
+
+    if len(unpadding) != 2:
+        raise AssertionError("Unpadding must be either an int or a tuple of two ints")
+
+    ones = torch.ones(patches.shape[0], patches.shape[2], original_size[0], original_size[1])
```

Comment:
`ones_like` after patches, to infer device

Reply:
Not sure what you mean. Can you elaborate? `ones_like(input)` is equivalent to `ones(input.size())` but that's not what we're doing here,

Reply:
yes, sorry, the following should work, otherwise this code most likely will crash with cuda device
 `torch.ones(patches.shape[0], patches.shape[2], original_size[0], original_size[1], device=patches.device, dtype=patches.dtype)`